githubapp/client_creator.go

```diff
@@ -223,9 +240,34 @@ func (c *clientCreator) newV4Client(base *http.Client, details string) (*githubv
 	return client, nil
 }
 
-func applyMiddleware(base *http.Client, middleware []ClientMiddleware) {
+func (c *clientCreator) transport() http.RoundTripper {
+	if c.cacheFunc != nil {
+		return cacheControl(c.cacheControlEnabled)(httpcache.NewTransport(c.cacheFunc()))
```

Comment:
Should be refactored to better handle using the 'installation middleware' approach. So far I've been unsuccessful coming up with the right pattern, so this is here as a poc.